### PR TITLE
Fix SignalFx receiver response

### PR DESCRIPTION
**Pull request by pjanotti**
The response expected is actually a JSON body.

### CODE REVIEW

File: receiver/signalfxreceiver/receiver.go [hunk 1]
```diff
@@ -253,3 +271,12 @@ func (r *sfxReceiver) failRequest(
 		zap.Error(err), // It handles nil error
 		zap.String("receiver", r.config.Name()))
 }
+
+func jsonResponse(s string) []byte {
+	respBody, err := json.Marshal(s)
+	if err != nil {
+		// This is to be used in initialization so panic here is fine.
```
Comment (owais) [inline receiver/signalfxreceiver/receiver.go]: nit: can we call it initJSONResponse or something similar to make the intention clear? Another idea is to use the init() function to make it very explicit. I generally don't like the init func but I guess in this case is acceptable. Your call.

File: receiver/signalfxreceiver/receiver.go [hunk 2]
```diff
@@ -232,6 +247,9 @@ func (r *sfxReceiver) failRequest(
 		}
 	}
 
+	// Use the same pattern as strings.Builder String().
+	msg := *(*string)(unsafe.Pointer(&jsonResponse))
```
Comment (owais) [inline receiver/signalfxreceiver/receiver.go]: Looks scary :) what does it do?

File: receiver/signalfxreceiver/receiver.go (hunk 1 above)
Comment (pjanotti) [reply receiver/signalfxreceiver/receiver.go]: I like rename but since is for private vars on the package I won't add a init().

**Review by tigrannajaryan — APPROVED** [review]

File: receiver/signalfxreceiver/receiver.go (hunk 2 above)
Comment (pjanotti) [reply receiver/signalfxreceiver/receiver.go]: avoids copying the contents of the `[]byte` it is stolen from here: https://golang.org/src/strings/builder.go?s=1379:1412#L45

**Review by owais — APPROVED** [review]

**Pull request merged**